testdata/sqllogictests_object_store/local/bson.slt

```diff
@@ -0,0 +1,33 @@
+statement ok
+COPY (select * from '${PWD}/testdata/parquet/userdata1.parquet') to '${TMP}/userdata1.bson';
+
+query I
+SELECT count(*) FROM '${TMP}/userdata1.bson';
+----
+1000
+
+statement ok
+create table timestamps (t timestamp);
+
+statement ok
+insert into timestamps values (arrow_cast(946684860000000, 'Timestamp(Microsecond, None)'));
+
+query
+select * from timestamps;
+----
+2000-01-01 00:01:00
+
+statement ok
+copy timestamps to '${TMP}/timestamp_test_out.bson';
+
+# bson's date format is an int64, of milliseconds since unix epoch;
```

Comment:
isn't there the 
> "UTC datetime - The int64 is UTC milliseconds since the Unix epoch." 

which isn't part of the internal mongodb spec


https://bsonspec.org/spec.html

Reply:
yes, exactly, that's what we're using. We just convert all arrow `Date` and `Timestamp` values into these int64 (millis since epoch), but when we read bson data back into GlareDB all of those values become `Date64`; not back to timestamps. 

```sql
> create table timestamps (t timestamp);
Table created
> insert into timestamps values (arrow_cast(946684860000000, 'Timestamp(Microsecond, None)'));
Inserted 1 row
>
>
> select * from timestamps;
┌─────────────────────┐
│ t                   │
│ ──                  │
│ Timestamp<µs, UTC>  │
╞═════════════════════╡
│ 2000-01-01T00:01:00 │
└─────────────────────┘
> copy timestamps to 'timestamps.bson';
Copy success
> select * from timestamps;
┌─────────────────────┐
│ t                   │
│ ──                  │
│ Timestamp<µs, UTC>  │
╞═════════════════════╡
│ 2000-01-01T00:01:00 │
└─────────────────────┘
> select * from 'timestamps.bson';
┌─────────────────────┐
│ t                   │
│ ──                  │
│ Date64              │
╞═════════════════════╡
│ 2000-01-01T00:01:00 │
└─────────────────────┘
```

It works as intended but `sqllogictest` renders `Date64` as `int64`s.

When we add support for declaring a schema rather than inferring it, we'll be able to cast these bson dates to whatever we need (with only the expected loss of fidelity and resolution), and the code _mostly_ supports this today.